In [5]:
# Set Variables
WKDIR=/workspace/hraijc/HiC_trials/Manuka
ASSEMBLY=${WKDIR}/primary.genome.scf.fasta
BAMFILE=${WKDIR}/manuka_V1_dedup.bam
APREFIX=manuka_V1
cd ${WKDIR}

In [6]:
mkdir -p ${WKDIR}/log

In [7]:
# Starting with Bamfile generated with HiC-scaffolding-pipeline that keeps stalling.
ln -s ${WKDIR}/hic-scaffolding-pipeline/results/bam/manuka_V1_dedup.bam ${BAMFILE}


In [8]:
#Just run YAHS

sbatch << EOF
#!/bin/bash
#SBATCH -J YAHS
#SBATCH -o ${WKDIR}/log/%J.out
#SBATCH -e ${WKDIR}/log/%J.err
#SBATCH --cpus-per-task=1
#SBATCH --mem=48G
#SBATCH --time=23:00:00

cd ${WKDIR}

### Run YAHS ###
#/workspace/hraijc/git_clones/yahs/yahs ${ASSEMBLY} ${BAMFILE} -o ${APREFIX}_yahs #This is actually _yahs_20k because it runs out of RAM silently. 
/workspace/hraijc/git_clones/yahs/yahs ${ASSEMBLY} ${BAMFILE} --no-mem-check -o ${APREFIX}_yahs_20K -r 20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000
/workspace/hraijc/git_clones/yahs/yahs ${ASSEMBLY} ${BAMFILE} --no-mem-check -o ${APREFIX}_yahs_10K -r 10000,20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000
/workspace/hraijc/git_clones/yahs/yahs ${ASSEMBLY} ${BAMFILE} --no-mem-check -o ${APREFIX}_yahs_5K -r 5000,10000,20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000
/workspace/hraijc/git_clones/yahs/yahs ${ASSEMBLY} ${BAMFILE} --no-mem-check -o ${APREFIX}_yahs_1K -r 1000,5000,10000,20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000




### YAHS without error correction. ###
#/workspace/hraijc/git_clones/yahs/yahs --no-contig-ec --no-scaffold-ec $ASSEMBLY ${BAMFILE} -o ${APREFIX}_yahsNoEC




EOF


Submitted batch job 1543621


In [7]:
seff 1490847 # Running YAHS with 5G RAM
seff 1495673 # Running YAHS with 24G RAM # Still not using the RAM....
seff 1513064 # Running YAHS with 48G RAM and --no-mem-check. Trying different HiC resolutions....

Job ID: 1490847
Cluster: powerplant
User/Group: hraijc/hraijc
State: FAILED (exit code 127)
Cores: 1
CPU Utilized: 02:15:22
CPU Efficiency: 98.50% of 02:17:26 core-walltime
Job Wall-clock time: 02:17:26
Memory Utilized: 2.48 GB
Memory Efficiency: 49.60% of 5.00 GB
Job ID: 1495673
Cluster: powerplant
User/Group: hraijc/hraijc
State: COMPLETED (exit code 0)
Cores: 1
CPU Utilized: 02:52:58
CPU Efficiency: 96.61% of 02:59:02 core-walltime
Job Wall-clock time: 02:59:02
Memory Utilized: 2.48 GB
Memory Efficiency: 10.33% of 24.00 GB
Job ID: 1513064
Cluster: powerplant
User/Group: hraijc/hraijc
State: COMPLETED (exit code 0)
Cores: 1
CPU Utilized: 07:03:13
CPU Efficiency: 99.61% of 07:04:52 core-walltime
Job Wall-clock time: 07:04:52
Memory Utilized: 22.30 GB
Memory Efficiency: 46.45% of 48.00 GB


In [2]:
out=${APREFIX}_yahs_5K # prefix of outfiles produced by YAHS.  
contigs=${ASSEMBLY} # need to be indexed, i.e., test.fa.gz.fai in same directory
wkdir=${WKDIR}


sbatch --job-name=hic_mapyahs \
    -o ${WKDIR}/log/%J.out \
    -e ${WKDIR}/log/%J.err \
    --cpus-per-task=8 \
    --mem=24G \
    --time=03:10:00 \
    --export=out=$out,contigs=$contigs,wkdir=$wkdir \
    /workspace/hraijc/Gitrepos/High-quality-genomes/Methods/DNase_HiC/notebooks/yahs_contactmapgen2.sh

Submitted batch job 1527616


In [5]:
out=${APREFIX}_yahs_10K # prefix of outfiles produced by YAHS.  
contigs=${ASSEMBLY} # need to be indexed, i.e., test.fa.gz.fai in same directory
wkdir=${WKDIR}


sbatch --job-name=hic_mapyahs \
    -o ${WKDIR}/log/%J.out \
    -e ${WKDIR}/log/%J.err \
    --cpus-per-task=8 \
    --mem=24G \
    --time=03:10:00 \
    --export=out=$out,contigs=$contigs,wkdir=$wkdir \
    /workspace/hraijc/Gitrepos/High-quality-genomes/Methods/DNase_HiC/notebooks/yahs_contactmapgen2.sh

Submitted batch job 1527736


In [3]:
out=${APREFIX}_yahs_20K # prefix of outfiles produced by YAHS.  
contigs=${ASSEMBLY} # need to be indexed, i.e., test.fa.gz.fai in same directory
wkdir=${WKDIR}


sbatch --job-name=hic_mapyahs \
    -o ${WKDIR}/log/%J.out \
    -e ${WKDIR}/log/%J.err \
    --cpus-per-task=8 \
    --mem=24G \
    --time=03:10:00 \
    --export=out=$out,contigs=$contigs,wkdir=$wkdir \
    /workspace/hraijc/Gitrepos/High-quality-genomes/Methods/DNase_HiC/notebooks/yahs_contactmapgen2.sh

Submitted batch job 1539915


In [ ]:
##############
#Do manual scaffolding cleanup in Juicer then run

##############

In [ ]:
#Just run YAHS

sbatch << EOF
#!/bin/bash
#SBATCH -J YAHS
#SBATCH -o ${WKDIR}/log/%J.out
#SBATCH -e ${WKDIR}/log/%J.err
#SBATCH --cpus-per-task=1
#SBATCH --mem=4G
#SBATCH --time=2:00:00

module load samtools


cd ${WKDIR}

module load samtools

/workspace/hraijc/git_clones/yahs/juicer post -o ${OUT_PREFIX}_v1 ${OUT_PREFIX}_JBAT.review.assembly ${OUT_PREFIX}_JBAT.liftover.agp ${ASSEMBLY_PRESCAFF}



EOF

In [4]:



cd ${WKDIR}
module load BBMap/38.33

statswrapper.sh in=$ASSEMBLY,${WKDIR}/${APREFIX}_yahs_5K_scaffolds_final.fa,${WKDIR}/${APREFIX}_yahs_10K_scaffolds_final.fa,${WKDIR}/${APREFIX}_yahs_20K_scaffolds_final.fa format=3 
module unload BBMap/38.33

java -ea -Xmx200m -cp /software/bioinformatics/BBMap-38.33/current/ jgi.AssemblyStatsWrapper format=3 in=/workspace/hraijc/HiC_trials/HiC23/Assemblies/Crodergsii_HiFi130423_purged.fa,/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_HiC070623_yahs_5K_scaffolds_final.fa,/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_HiC070623_yahs_10K_scaffolds_final.fa,/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_HiC070623_yahs_20K_scaffolds_final.fa format=3
n_scaffolds	n_contigs	scaf_bp	contig_bp	gap_pct	scaf_N50	scaf_L50	ctg_N50	ctg_L50	scaf_N90	scaf_L90	ctg_N90	ctg_L90	scaf_max	ctg_max	scaf_n_gt50K	scaf_pct_gt50K	gc_avg	gc_std	filename
2770	2887	883753165	883750474	0.000	498	559676	501	557004	1595	156350	1621	150094	4267868	4267868	2376	98.531	0.39359	0.00860	/powerplant/workspace/hraijc/HiC_trials/HiC23/Assemblies/Crodergsii_HiFi130423_purged.fa
226	3002	884284965	883750474	0.060	7	40653293	515	549938	17	31167351	1661	146243	92137127	4267868	123	99.762	0.39

| n_scaffolds 	| n_contigs 	| scaf_bp   	| contig_bp 	| gap_pct 	| scaf_N50 	| scaf_L50 	| ctg_N50 	| ctg_L50 	| scaf_N90 	| scaf_L90 	| ctg_N90 	| ctg_L90 	| scaf_max  	| ctg_max 	| scaf_n_gt50K 	| scaf_pct_gt50K 	| gc_avg  	| gc_std  	| filename                                         	|
|-------------	|-----------	|-----------	|-----------	|---------	|----------	|----------	|---------	|---------	|----------	|----------	|---------	|---------	|-----------	|---------	|--------------	|----------------	|---------	|---------	|--------------------------------------------------	|
| 2770        	| 2887      	| 883753165 	| 883750474 	| 0.000   	| 498      	| 559676   	| 501     	| 557004  	| 1595     	| 156350   	| 1621    	| 150094  	| 4267868   	| 4267868 	| 2376         	| 98.531         	| 0.39359 	| 0.00860 	| Crodergsii_HiFi130423_purged.fa                  	|
| 1154        	| 3002      	| 884099365 	| 883750474 	| 0.039   	| 11       	| 35672154 	| 515     	| 549938  	| 55       	| 582957   	| 1661    	| 146243  	| 65081279  	| 4267868 	| 693          	| 98.408         	| 0.39359 	| 0.01363 	| Crodergsii_HiC070623_yahs_scaffolds_final.fa     	|
| 536         	| 2887      	| 884199965 	| 883750474 	| 0.051   	| 7        	| 44190339 	| 501     	| 557004  	| 17       	| 23979814 	| 1621    	| 150094  	| 139140493 	| 4267868 	| 234          	| 98.999         	| 0.39359 	| 0.01326 	| Crodergsii_HiC070623_yahsNoEC_scaffolds_final.fa 	|


| n_scaffolds 	| n_contigs 	| scaf_bp   	| contig_bp 	| gap_pct 	| scaf_N50 	| scaf_L50 	| ctg_N50 	| ctg_L50 	| scaf_N90 	| scaf_L90 	| ctg_N90 	| ctg_L90 	| scaf_max 	| ctg_max 	| scaf_n_gt50K 	| scaf_pct_gt50K 	| gc_avg  	| gc_std  	| filename                                                                                                      	|
|-------------	|-----------	|-----------	|-----------	|---------	|----------	|----------	|---------	|---------	|----------	|----------	|---------	|---------	|----------	|---------	|--------------	|----------------	|---------	|---------	|---------------------------------------------------------------------------------------------------------------	|
| 2770        	| 2887      	| 883753165 	| 883750474 	| 0.000   	| 498      	| 559676   	| 501     	| 557004  	| 1595     	| 156350   	| 1621    	| 150094  	| 4267868  	| 4267868 	| 2376         	| 98.531         	| 0.39359 	| 0.00860 	| /powerplant/workspace/hraijc/HiC_trials/HiC23/Assemblies/Crodergsii_HiFi130423_purged.fa                      	|
| 226         	| 3002      	| 884284965 	| 883750474 	| 0.060   	| 7        	| 40653293 	| 515     	| 549938  	| 17       	| 31167351 	| 1661    	| 146243  	| 92137127 	| 4267868 	| 123          	| 99.762         	| 0.39359 	| 0.01995 	| /powerplant/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_HiC070623_yahs_5K_scaffolds_final.fa  	|
| 337         	| 3002      	| 884262765 	| 883750474 	| 0.058   	| 9        	| 39820943 	| 515     	| 549938  	| 18       	| 32836090 	| 1661    	| 146243  	| 76945733 	| 4267868 	| 167          	| 99.575         	| 0.39359 	| 0.01835 	| /powerplant/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_HiC070623_yahs_10K_scaffolds_final.fa 	|
| 620         	| 3002      	| 884206165 	| 883750474 	| 0.052   	| 9        	| 39310224 	| 515     	| 549938  	| 19       	| 27936322 	| 1661    	| 146243  	| 98960227 	| 4267868 	| 253          	| 98.885         	| 0.39359 	| 0.01624 	| /powerplant/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_HiC070623_yahs_20K_scaffolds_final.fa 	|
|

## Appendix

The dataset was passed through PFR's HiC-Scaffolding-Pipeline.

To do it yourself see below:


In [ ]:
#Set Variables
SPDIR=${WKDIR}/Karaka
mkdir -p $SPDIR
ASSEMBLY=${GENOMEDIR}/karaka_phasing.hic.hap1.p_ctg.fasta
APREFIX=Karaka_hic
HiC_RAW=${HiCDIR}/Karaka-5-Hi-C_S3_L001_R
READ1=${HiC_RAW}1_001.fastq.gz
READ2=${HiC_RAW}2_001.fastq.gz
TEMP=${WKDIR}/TEMP

In [ ]:
#Run HiC workflow

sbatch << EOF
#!/bin/bash
#SBATCH -J hic
#SBATCH -o ${WKDIR}/log/%J.out
#SBATCH -e ${WKDIR}/log/%J.err
#SBATCH --cpus-per-task=12
#SBATCH --mem=10G
#SBATCH --time=5:00:00

module load bwa/0.7.17
module load samtools/1.16
module load conda
module load fastp/0.23.2


### Clean HiC Reads with fastp.###
fastp \
-i ${READ1} \
-o ${HiC_CLEAN}/${APREFIX}_Clean_R1.fastq.gz \
-I ${READ2} \
-O ${HiC_CLEAN}/${APREFIX}_Clean_R2.fastq.gz \
#--trim_front1 15 \ # Legacy cleaning parameter. Don't think it improves output. 
#--trim_front2 15 \
--qualified_quality_phred 20 \
--length_required 50 \
--thread 12


###Align reads with BWA. Use -5SP for Hi-C reads.###
bwa index $ASSEMBLY
bwa mem -5SP -t12 ${ASSEMBLY} ${HiC_CLEAN}/${APREFIX}_Clean_R1.fastq.gz ${HiC_CLEAN}/${APREFIX}_Clean_R2.fastq.gz -o ${TEMP}/${APREFIX}.sam



###Flag PCR Duplicates with SAMBLASTER###
/workspace/hraijc/git_clones/samblaster/samblaster -i ${TEMP}/${APREFIX}.sam -o ${TEMP}/${APREFIX}_marked_byread.sam



### Remove unmmaped and non-primary aligned reads. Sort and Index bam files.### 
samtools view -S -b -h -@ 12 -F 2316 ${TEMP}/${APREFIX}_marked_byread.sam > ${TEMP}/${APREFIX}_presort_marked.bam

samtools sort -@ 12 ${TEMP}/${APREFIX}_presort_marked.bam -o ${APREFIX}_marked.bam
samtools sort -@ 12 -n ${TEMP}/${APREFIX}_presort_marked.bam -o ${APREFIX}_marked_namesort.bam

samtools view -b -h -@ 12 -F 3340 ${APREFIX}_marked.bam > ${TEMP}/${APREFIX}_presort_marked.bam 

samtools sort -@ 12 ${TEMP}/${APREFIX}_presort_marked.bam -o ${APREFIX}_dedup.bam
#samtools sort -@ 12 -n ${TEMP}/${APREFIX}_presort_marked.bam -o ${APREFIX}_dedup_namesort.bam

#samtools index -@ 12 ${APREFIX}_marked.bam 

samtools index -@ 12 ${APREFIX}_dedup.bam 

#samtools flagstat -@ 12 ${APREFIX}_marked.bam > flagstat.txt
samtools flagstat -@ 12 ${APREFIX}_dedup.bam  > flagstat_dedup.txt


### HiC QC from Phase genomics ###
conda activate hraijc_hic_qc2
python /workspace/hraijc/git_clones/hic_qc/hic_qc.py -n -1 -b ${APREFIX}_marked_namesort.bam --outfile_prefix ${APREFIX}_marked
python /workspace/hraijc/git_clones/hic_qc/hic_qc.py -n -1 -b ${APREFIX}_dedup_namesort.bam --outfile_prefix ${APREFIX}_dedup

conda deactivate


## Viz HiC Coverage #####
#/home/hraijc/.local/bin/plotCoverage -b ${APREFIX}_marked.bam --plotFile example_coverage -n 100000 -o HiC_Coverage_allreads.png --labels ${APREFIX} --plotTitle "HiC_Coverage all reads" 
#/home/hraijc/.local/bin/plotCoverage -b ${APREFIX}_dedup.bam --plotFile example_coverage -n 100000 -o HiC_Coverage_unique_reads.png --labels ${APREFIX} --plotTitle "HiC_Coverage PCR duplicates removed" 


### Run YAHS ###
###SEE ABOVE


EOF
